pkg/metrics/wal/series.go

```diff
@@ -216,6 +226,24 @@ func (s *stripeSeries) set(hash uint64, series *memSeries) {
 	s.locks[i].Unlock()
 }
 
+func (s *stripeSeries) getLatestExemplar(id uint64) *exemplar.Exemplar {
+	i := id & uint64(s.size-1)
+
+	s.locks[i].RLock()
+	exemplar := s.exemplars[i][id]
+	s.locks[i].RUnlock()
+
+	return exemplar
+}
+
+func (s *stripeSeries) setLatestExemplar(id uint64, exemplar *exemplar.Exemplar) {
+	i := id & uint64(s.size-1)
+
+	s.locks[i].Lock()
+	s.exemplars[i][id] = exemplar
```

Comment:
nit: this has a very low chance of leaking if there's no series for `id`. Maybe this should check to see if `s.series[i][id]` is non-nil?

Reply:
Yeah, I hadn't considered this since supposedly we garbage collect keys from both series and exemplars in one pass, but better to seal this.